managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java

```diff
@@ -55,6 +66,53 @@ public void run() {
         universe = lockUniverseForUpdate(-1 , true);
       }
 
+      if (params().isDeleteBackups) {
+        List<Backup> backupList = Backup.fetchByUniverseUUID(params().customerUUID, universe.universeUUID);
+        for (Backup backup : backupList) {
+          try {
+            backup = Backup.get(params().customerUUID, backup.backupUUID);
+            if (backup.state != Backup.BackupState.Completed) {
+              LOG.error("Cannot delete backup in any other state other than completed.");
+            }
+            backup.transitionState(Backup.BackupState.Deleted);
+            BackupTableParams backupParams = Json.fromJson(backup.backupInfo, BackupTableParams.class);
+            if (backupParams.backupList != null) {
+              for (BackupTableParams childBackupParams : backupParams.backupList) {
+                childBackupParams.actionType = BackupTableParams.ActionType.DELETE;
+                ShellResponse response = tableManager.deleteBackup(childBackupParams);
+                JsonNode jsonNode = Json.parse(response.message);
+                if (response.code != 0 || jsonNode.has("error")) {
+                  // Revert state to completed since it couldn't get deleted.
+                  backup.transitionState(Backup.BackupState.Completed);
+                  LOG.error("Delete Backup failed for {}. Response code={}, hasError={}.",
+                            childBackupParams.storageLocation, response.code, jsonNode.has("error"));
```

Comment:
The same here - do we need `break` here? Because you are updating state of the whole backup...

Reply:
yeah, but I am not sure what this part of code doing. @Arnav15 can you please help me to understand this?
Is this related to backing up multiple tables? ( entire universe )

Reply:
Yes, this is if there are multiple backup objects as part of a single backup (Universe, keyspace, namespace, etc., multi-table)